src/main/java/build/buildfarm/common/DigestUtil.java

```diff
@@ -96,34 +97,13 @@ public DigestUtil(HashFunction hashFn) {
   }
 
   public Digest compute(ByteString blob) {
+    Hasher hasher = hashFn.getHash().newHasher();
```

Comment:
Nice, I need to do the same in Bazel code!

Reply:
I really wanted to get to being able to treat the sink as a legit sink without the possibility of IOException due to the InputStream round trip, but my eyes glassed over as I pulled apart hasher code. lmk if you come up with something better for bazel!